STM32/Util/Src/systeminfo.c

```diff
@@ -205,6 +207,13 @@ const char* statusInfo(bool printStart)
             (int)bt, (int)BS.boardType, pv.major, pv.minor, BS.pcb_version.major, BS.pcb_version.minor);
     }
 
+    if (BS.usb)
+    {
+        len += snprintf(&buf[len], sizeof(buf) - len, 
```

Comment:
I think this will be useful and in any case much better than the lack of fault handling in place at the moment.

Only pitfall would be that often times when boards have communication errors they will be power cycled. In this instance it will be hard to know what caused an error. 

I will admit that persistent errors are probably due to some other fault, but nevertheless I expect a portion of this diagnostic will be lost.

Reply:
Agreed. It is hard to know precisely what to do in the case the USB is down